uint64 last_received
uint64[] missing_msgs
